### PR TITLE
use partitioned update on truncating tables

### PR DESCRIPTION
**Pull request by iwata**
## WHAT
SSIA

## WHY
to avoid the limit of 20,000 mutations

### CODE REVIEW

File: pkg/spanner/client.go [hunk 1]
```diff
@@ -143,13 +143,13 @@ func (c *Client) TruncateAllTables(ctx context.Context) error {
...
 
-	_, err = c.spannerClient.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
-		return txn.BufferWrite(m)
-	})
-	if err != nil {
-		return &Error{
-			Code: ErrorCodeTruncateAllTables,
-			err:  err,
+	for _, stmt := range stms {
+		_, err := c.spannerClient.PartitionedUpdate(ctx, stmt)
```
Comment (110y) [inline pkg/spanner/client.go:147]: It's better to make this invocation parallel with each other.

Comment (iwata) [reply pkg/spanner/client.go]: fixed 5e1dea0

**Comment by iwata** [comment]
@110y PTAL

File: pkg/spanner/client.go [hunk 2]
```diff
@@ -143,10 +144,15 @@ func (c *Client) TruncateAllTables(ctx context.Context) error {
...
 	}
 
-	_, err = c.spannerClient.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
-		return txn.BufferWrite(m)
-	})
-	if err != nil {
+	eg := errgroup.Group{}
+	for _, stmt := range stms {
+		stmt := stmt
+		eg.Go(func() error {
```
Comment (110y) [inline pkg/spanner/client.go:150]: With `errgroup.Group`, we can recognize only the first error returned by eg.Go's argument (function). So, it's better to handle all errors by using multierror: https://github.com/hashicorp/go-multierror

Comment (iwata) [reply pkg/spanner/client.go]: fixed and used `sync.WaitGroup` instead of `errgroup.Group`.

File: pkg/spanner/client.go [hunk 3]
```diff
@@ -143,13 +145,29 @@ func (c *Client) TruncateAllTables(ctx context.Context) error {
 		}
 	}
 
-	_, err = c.spannerClient.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
-		return txn.BufferWrite(m)
-	})
-	if err != nil {
+	var (
+		wg     = &sync.WaitGroup{}
```
Comment (110y) [inline pkg/spanner/client.go:149]: we should use multierrors' `Group` instead sync.WaitGroup https://pkg.go.dev/github.com/hashicorp/go-multierror#Group

Comment (iwata) [reply pkg/spanner/client.go:149]: That makes sense!
I'll try it, soon...

Comment (iwata) [reply pkg/spanner/client.go]: done!

**Comment by 110y** [comment]
@iwata Thanks for contribution!

**Review by 110y — APPROVED** [review]

**Pull request merged**